frontend-ssr/pages/submissions.tsx

```diff
@@ -22,22 +22,49 @@ interface State {
 }
 
 class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private hiddenRef = React.createRef<HTMLDivElement>();
+
   constructor(props) {
     super(props);
 
     this.state = {
-      loading: props.submissions.submissions === undefined,
+      loading: false,
     };
   }
 
   componentDidMount() {
-    let { submissions } = this.props;
+    this.initIntersectionObserver();
```

Comment:
You should only initialize it if we haven't reached the end.  Otherwise, whenever you go back-forward, it will try and fetch the next page.

Reply:
Also, make sure the `disconnect` inside `componentWillUnmount` also check if this observer has been initialized.